### PR TITLE
[#18][#21][Part2] [UI] Survey Details screen

### LINKED ISSUE #18: [UI] As a user, I can see the survey details when taking a survey
## Why

App must display the survey questions screen when taking a survey

## Acceptance Criteria

- Must display `Survey.coverImageUrl` as background
- Must display question index and total question count
  - do not include intro question
- If first question, 
  - show `Take Survey` button
  - show back button 
- if last question
  - show `Submit Survey` button
  - show close button
- if not first or last question
  - display Floating Action Button with right chevron icon
  - show close button

## Resources

- [Design](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=34%3A868)

### LINKED ISSUE #21: [UI] As a user, I can see the intro question when taking a survey
## Why

App must display the intro question before taking a survey

## Acceptance Criteria

- Must display the intro question
- Title should be `Survey.title`
- subtitle should be `Question.text`

## Resources

- [Design](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=34%3A868)

### PR DESCRIPTION
**Pull request by lydiasama**
- Closes #18 
- Closes #21 

## What happened 👀

- [x] [Part 1]: Set up AppNavigation. 

- [x] [Part 2]: Implement UI for Survey Details screen

## Insight 📝

- Added `QuestionUiModel`.
- Updated the `RoundedRectangleButtonWidget` with wrap_content width as default to reuse on the question screens.
- Set `physics: const NeverScrollableScrollPhysics(),` [here](https://github.com/nimblehq/flutter-ic-lydia-ryan/compare/feature/18-ui-survey-details...feature/18-part2-ui-survey-details?expand=1#diff-c2055110ded419d42cb58ddbd1a4ec723cc53b414f860d7a9a53fa3dd01e1958R37) to avoid slide left to change page.
- Navigate back to `HomeScreen` when clicking on `back` and `close` icon.

## Proof Of Work 📹

https://user-images.githubusercontent.com/12026942/234250118-b5de9525-2531-412a-94d0-bfa481a6a644.mp4

### CODE REVIEW

File: lib/resources/dimensions.dart [hunk 1]
```diff
@@ -15,6 +16,7 @@ class Dimensions {
 
   // Button
   static const double buttonHeight = 56.0;
+  static const double buttonHorizontalPadding = 21;
```
Comment (markgravity) [inline lib/resources/dimensions.dart]: We are missing the leading zero here 🙏 
`21` -> `21.0`

File: lib/screens/surveydetails/widget/question_paging_widget.dart [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+  }
+
+  Widget _submitSurveyButtonWidget(BuildContext context) {
+    return RoundedRectangleButtonWidget(
+      text: AppLocalizations.of(context)!.submit,
+      onPressed: () {},
+    );
+  }
+
+  bool _isIntroQuestion(QuestionUiModel question) =>
```
Comment (markgravity) [inline lib/screens/surveydetails/widget/question_paging_widget.dart:198]: What do you think about moving these properties along with `int _currentIndex = 0;` on the top? 🤔

**Review by markgravity — COMMENTED** [review]
The rest LGTM

File: lib/screens/surveydetails/widget/question_paging_widget.dart (hunk 2 above)
Comment (lydiasama) [reply lib/screens/surveydetails/widget/question_paging_widget.dart:198]: @markgravity They're not properties. They're functions. 🤔  Should we place them to the bottom after Widget function?

Comment (markgravity) [reply lib/screens/surveydetails/widget/question_paging_widget.dart:198]: @lydiasama my bad, I mistake with the getter function. You're good! 👍

**Review by markgravity — APPROVED** [review]

**Pull request merged**